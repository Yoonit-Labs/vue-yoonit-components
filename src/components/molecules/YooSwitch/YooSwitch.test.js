import { shallowMount } from '@vue/test-utils'
import YooSwitch from '@/components/molecules/YooSwitch/YooSwitch.vue'
import PropsConfig from './YooSwitch.config'

const classBlock = 'yoo-switch'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooSwitch, {
    slots: { default: SlotText }
  })
}

describe('YooSwitch Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', async () => {
    await wrapper.setData({ timeId: 'custom-time-id' })
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.classes('yoo-switch-container')).toBe(true)
    expect(wrapper.find('input.yoo-switch').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('size', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.size.options.includes(YooSwitch.props.size.default)).toBe(true)
      })
      PropsConfig.size.options.forEach(size => {
        it(`Includes size class: .${classBlock}--${size}`, async () => {
          await wrapper.setProps({ size })
          expect(wrapper.find(`.${classBlock}--${size}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props
  })
