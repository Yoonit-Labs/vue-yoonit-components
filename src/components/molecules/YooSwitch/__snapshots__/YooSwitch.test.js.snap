// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooSwitch Component Matches Snapshot 1`] = `<div class="yoo-switch-container m__t--xl m__b--xl m__r--xs m__l--xs"><input type="checkbox" class="yoo-switch yoo-switch--medium" id="ycid-3456789"><label class="yoo-switch__label" for="ycid-3456789"></label></div>`;
